packages/shared/lib/core/ledger/utils/determineLedgerDeviceState.ts

```diff
@@ -0,0 +1,26 @@
+import { LedgerStatus, LedgerConnectionState } from '../interfaces'
+import { LedgerAppName } from '../enums'
+
+export function determineLedgerDeviceState(status: LedgerStatus): LedgerConnectionState {
+    const { locked, connected, app } = status
+    if (locked) {
+        return LedgerConnectionState.Locked
+    } else {
+        switch (app?.name) {
+            default:
+                if (connected) {
+                    /**
+                     * NOTE: "BOLOS" is the name of the Ledger operating system and is
+                     * sometimes registered as an app.
+                     */
+                    return app?.name && app?.name !== LedgerAppName.BOLOS
```

Comment:
Ah, probably changing enum values will break this logic?

Reply:
Yep, I think we can keep it as is?